pkg/endpoint/local_ip.go

```diff
@@ -29,27 +33,42 @@ var familyToDestIP = map[k8snet.IPFamily]string{
 	k8snet.IPv6: "2001:4860:4860::8888",
 }
 
-func getLocalIPv4ForDestination(dst string) string {
-	conn, err := net.Dial("udp4", dst+":53")
-	logger.FatalOnError(err, "Error getting local IP")
+func getLocalIPFromRoutes(family string) (string, error) {
```

Comment:
Does this work for IPv6? Do we still need this with https://github.com/submariner-io/submariner/pull/3373/commits/24f1b964576c1676179b95425b8b2aa79093bd44?

Reply:
If you meant to GetLocalIP function, yeah we still need getLocalIPFromRoutes fallback, GetLocalIP is called regardless to air-gap status from [1] .

> Does this work for IPv6? 

getLocalIPFromRoutes returns the IP address of the local interface used in the default V4/V6 route, I think it should work also for IPv6, also noticed that dual-stack gate is green , so GetLocalIP for v6 doesn't crash.

[1] 
https://github.com/submariner-io/submariner/blob/devel/pkg/endpoint/local_endpoint.go#L182

Reply:
Hmm, getLocalIPFromRoutes looks fine in e2e dual-stack but it fails in [ local endpoint unit test with dual-stack](https://github.com/submariner-io/submariner/blob/devel/pkg/endpoint/local_endpoint_test.go#L180) , I added getLocalIPFromRoutes  hook to workaround it.

Reply:
`getLocalIPFromRoutes` uses the netlink API wrapper so, as long as the fake netlink implementation is hooked in and mocked, then it should work. But I can take a look at that in a follow-up. Let's focus on getting the production working and merging this.